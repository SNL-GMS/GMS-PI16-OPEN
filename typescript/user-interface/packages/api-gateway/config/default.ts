// To set mock enabled false run with MOCK=false or MOCK=FALSE
// all other values for MOCK ('undefined', 'true', 't', '', random string) will result in mock true
const mockEnabled =
  process.env.MOCK === undefined ||
  process.env.MOCK === '' ||
  process.env.MOCK.toLocaleLowerCase() !== 'false';

// Default Config
const defaultConfig = {
  configName: 'default',
  logLevel: process.env.LOG_LEVEL || 'info',
  inMemorySession: true,

  // Server configuration
  server: {
    graphql: {
      http: {
        host: 'localhost',
        port: 3000,
        graphqlPath: '/interactive-analysis-api-gateway/graphql'
      },
      ws: {
        host: 'localhost',
        port: 4000,
        path: '/interactive-analysis-api-gateway/subscriptions'
      }
    }
  },

  // Kafka configuration
  kafka: {
    clientId: 'api-gateway',
    groupId: 'user-interface',
    brokers: ['kafka:9092'],
    connectionTimeout: 3000,
    maxWaitTimeInMs: 100,
    heartbeatInterval: 500, // ms
    consumerTopics: {
      systemMessagesTopic: 'system.system-messages',
      uiStationSoh: 'soh.ui-materialized-view'
    },
    producerTopics: {
      acknowledgedTopic: 'soh.ack-station-soh',
      quietedTopic: 'soh.quieted-list'
    }
  },

  // Configuration configuration
  configuration: {
    resolvers: {},
    subscriptions: {
      channels: {}
    },
    backend: {
      mock: {
        enable: mockEnabled,
        serviceDelayMs: 0
      },
      // Service endpoints for this component
      services: {
        getAnalystConfiguration: {
          requestConfig: {
            method: 'post',
            url: `http://ui-processing-configuration-service:8080/ui-processing-configuration-service/resolve`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'application/json',
              'content-type': 'application/json'
            },
            timeout: 60000
          }
        },
        getSohConfiguration: {
          requestConfig: {
            method: 'post',
            url: `http://ssam-control:8080/ssam-control/retrieve-station-soh-monitoring-ui-client-parameters`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'text/plain',
              'content-type': 'text/plain'
            },
            timeout: 120000
          }
        }
      }
    }
  },

  // Performance monitoring configuration
  performanceMonitoring: {
    resolvers: {},
    subscriptions: {
      channels: {
        sohStatus: 'sohStatus'
      }
    },
    backend: {
      mock: {
        enable: mockEnabled,
        serviceDelayMs: 0
      },
      // Service endpoints for this component
      services: {
        getHistoricalSohData: {
          requestConfig: {
            method: 'post',
            url: `http://ssam-control:8080/ssam-control/retrieve-decimated-historical-station-soh`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'application/json',
              'content-type': 'application/json'
            },
            timeout: 120000
          }
        },
        getHistoricalAceiData: {
          requestConfig: {
            method: 'post',
            url: `http://frameworks-osd-service:8080/frameworks-osd-service/osd/coi/acquired-channel-environment-issues/query/station-id-time-and-type`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'application/json',
              'content-type': 'application/json'
            },
            timeout: 120000
          }
        }
      }
    }
  },

  // Processing Station configuration
  processingStation: {
    resolvers: {},
    subscriptions: {
      channels: {}
    },
    backend: {
      mock: {
        enable: mockEnabled,
        serviceDelayMs: 0
      },
      // Service endpoints for this component
      services: {
        stationGroupByName: {
          requestConfig: {
            method: 'post',
            url: `http://frameworks-osd-service:8080/frameworks-osd-service/osd/station-groups`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'application/json',
              'content-type': 'application/json'
            },
            timeout: 120000
          }
        },
        stationsByNames: {
          requestConfig: {
            method: 'post',
            url: `http://frameworks-osd-service:8080/frameworks-osd-service/osd/stations`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'application/json',
              'content-type': 'application/json'
            },
            timeout: 120000
          }
        },
        channelsByNames: {
          requestConfig: {
            method: 'post',
            url: `http://frameworks-osd-service:8080/frameworks-osd-service/osd/channels`,
            responseType: 'json',
            proxy: false,
            headers: {
              accept: 'application/json',
              'content-type': 'application/json'
            },
            timeout: 120000
          }
        }
      }
    }
  },

  // System Messages Configuration
  systemMessage: {
    backend: {
      mock: {
        enable: mockEnabled
      },
      // Service endpoints for this component
      services: {
        getSystemMessageDefinitions: {
          requestConfig: {
            method: 'post',
            url: `http://smds-service:8080/smds-service/retrieve-system-message-definitions`,
            headers: {
              accept: 'text/plain',
              'content-type': 'text/plain'
            },
            proxy: false,
            timeout: 60000
          }
        }
      }
    },
    subscriptions: {
      systemMessages: 'systemMessages'
    }
  },

  // Other Info - test data files
  testData: {
    standardTestDataSet: {
      stdsDataHome: 'resources/test_data/unit-test-data/Standard_Test_Data',
      stdsJsonDir: 'gms_test_data_set',
      stationProcessing: {
        stationGroupsFileName: 'processing-station-group.json'
      }
    },
    // Test data not part of STDS
    additionalTestData: {
      dataPath: 'resources/test_data/additional-test-data/',
      systemMessageDefinitionsFileName: 'systemMessageDefinitions.json',
      filterChannelFileName: 'filterProcessingChannels.json',
      waveformFilterFileName: 'filterParameters.json',
      qcMaskFileName: 'qcMasks.json',
      uiConfigFileName: 'uiConfig.json',
      stationGroupsFileName: 'processing-station-group.json',
      stationGroupSohStatus: 'stationGroupSohStatus.json',
      stationSoh: 'stationSoh.json',
      stationSohOld: 'stationSohOld.json',
      transferredFileName: 'transferredFile.json',
      featurePredictionAzimuth: 'featurePredictionAzimuth.json',
      featurePredictionSlowness: 'featurePredictionSlowness.json',
      featurePredictionArrival: 'featurePredictionArrival.json',
      networkMagnitudeSolutions: 'networkMagnitudeSolutions.json',
      channelCalibrationFile: 'calibration.json',
      analystConfigurationFilename: 'analystConfiguration.json',
      commonConfigurationFilename: 'commonConfiguration.json',
      uiSohConfigurationFileName: 'ui-soh-settings.json',
      uiSohConfigurationOldFileName: 'ui-soh-settings-old.json',
      uiSohStationGroupFileName: 'soh.station-groups.json',
      sohControlFileName: 'soh-control.json',
      historicalSohFilename: 'historicalSohResponse.json',
      historicalAceiFilename: 'historicalAceiResponse.json'
    },
    // Inputs for integration tests - currently deprecated
    integrationInputs: {
      dataPath: 'resources/test_data/integration-inputs/',
      featurePrediction: 'feature-prediction-input.json',
      filterWaveform: 'filter-waveform-input.json',
      saveQcMask: 'save-qcmask-input.json',
      saveSignalDetection: 'save-sd-input.json',
      locateEvent: 'locate-event-input.json',
      computeFk: 'compute-fk-input.json',
      computeBeam: 'compute-beam-input.json',
      saveEvent: 'save-event-input.json',
      saveWaveformChannelSegment: 'save-waveform-channel-segment-input.json',
      saveFkChannelSegment: 'save-fk-channel-segment-input.json',
      computeMagSolution: 'compute-mag-solution.json'
    }
  },
  lateData: {
    detectionCount: 10,
    channelSegmentCount: 4,
    delayMillis: 10000,
    preStartDelay: 40000
  }
};

// eslint-disable-next-line import/no-default-export
export default defaultConfig;
