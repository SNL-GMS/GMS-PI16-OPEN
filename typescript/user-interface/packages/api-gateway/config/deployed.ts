import defaultConfig from './default';

// Overwrites default config for a deployed environment
const deployedConfig = {
  ...defaultConfig,

  configName: 'deployed',

  // session memory is from a DB not in memory
  inMemorySession: false,

  // Changes kafka brokers to be the deployed kafka containers
  kafka: {
    ...defaultConfig.kafka,
    brokers: ['kafka:9092']
  },

  configuration: {
    ...defaultConfig.configuration,
    backend: {
      ...defaultConfig.configuration.backend,
      mock: {
        enable: false
      }
    }
  },

  performanceMonitoring: {
    ...defaultConfig.performanceMonitoring,
    backend: {
      ...defaultConfig.performanceMonitoring.backend,
      mock: {
        enable: false
      }
    }
  },

  processingStation: {
    ...defaultConfig.processingStation,
    backend: {
      ...defaultConfig.processingStation.backend,
      mock: {
        enable: false
      }
    }
  },

  systemMessage: {
    ...defaultConfig.systemMessage,
    backend: {
      ...defaultConfig.systemMessage.backend,
      mock: {
        enable: false
      }
    }
  },

  // Change test data location for mock mode - deprecated?
  testData: {
    ...defaultConfig.testData,
    standardTestDataSet: {
      ...defaultConfig.testData.standardTestDataSet,
      stdsDataHome: '/opt/app-root/src/Test_Data_Sets/Standard_Test_Data/'
    }
  }
};

// eslint-disable-next-line import/no-default-export
export default deployedConfig;
