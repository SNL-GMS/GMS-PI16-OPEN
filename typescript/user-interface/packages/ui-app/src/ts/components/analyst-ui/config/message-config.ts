import { CacheTypes } from '@gms/common-model';

export interface MessageConfig {
  historyFilters: Map<string, boolean>;
  tooltipMessages: {
    history: {
      currentHistoryActionMessage: string;
      redoActionMessage: string;
      undoActionMessage: string;
      undoEventLevelActionMessage: string;
      redoEventLevelActionMessage: string;
      undoButtonAction: string;
      redoButtonAction: string;
    };
    location: {
      associatedOrCreatedMessage: string;
      locateCallInProgressMessage: string;
      rejectedOrUnassociatedMessage: string;
    };
    magnitude: {
      azimuthSourceToReceiverMessage: string;
      noStationsSetToDefiningMessage: string;
      setAllStationsNotDefiningMessage: string;
      setAllStationsDefiningMessage: string;
      sourceToReceiverAzimuthMessage: string;
      noAmplitudeMessage: string;
    };
    workflowConfirmation: {
      discardTooltip: string;
      cancelTooltip: string;
      header: string;
      text: string;
      title: string;
      cancelText: string;
      discardText: string;
    };
  };
}
export const messageConfig: MessageConfig = {
  // If true, filter out the keys from the history list
  historyFilters: new Map([
    [CacheTypes.UserActionDescription.UPDATE_EVENT_FROM_SIGNAL_DETECTION_CHANGE, true]
  ]),
  tooltipMessages: {
    history: {
      currentHistoryActionMessage: 'The current state displayed',
      redoActionMessage: 'Redo this action',
      undoActionMessage: 'Undo this action',
      undoEventLevelActionMessage: 'Revert event history',
      redoEventLevelActionMessage: 'Restore event history',
      undoButtonAction: 'Undo last action',
      redoButtonAction: 'Redo last undone action'
    },
    location: {
      associatedOrCreatedMessage: 'SD Associated or Created since last locate',
      locateCallInProgressMessage: 'Displays if a locate call is in progress',
      rejectedOrUnassociatedMessage: 'SD Rejected or Unassociated since last locate'
    },
    magnitude: {
      azimuthSourceToReceiverMessage: 'Source-to-Receiver Azimuth (\u00B0)',
      noStationsSetToDefiningMessage:
        'Select at least one defining station to calculate network magnitude',
      setAllStationsNotDefiningMessage: 'Set all stations as not defining',
      setAllStationsDefiningMessage: 'Set all stations as defining',
      sourceToReceiverAzimuthMessage: 'Source to Receiver Azimuth (\u00B0)',
      noAmplitudeMessage: 'Make amplitude measurement to set as defining'
    },
    workflowConfirmation: {
      discardTooltip: 'Discard your changes and open a new interval',
      cancelTooltip: 'Cancel and do not open a new interval',
      title: 'Warning',
      header: 'You have unsaved changes in your workspace.',
      text: ' Do you want to discard these changes and load data for a different time range?',
      cancelText: 'Cancel',
      discardText: 'Discard my changes'
    }
  }
};
