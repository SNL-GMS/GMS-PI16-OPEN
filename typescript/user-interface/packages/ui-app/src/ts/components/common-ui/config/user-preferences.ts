import { APP_PREFIX_PATH } from '@gms/common-util';

export interface UserPreferences {
  baseSoundsPath: string;
  tableRowHeightPx: number;
  soundFileNotFound(filename: string): string;
  configuredAudibleNotificationFileNotFound(filename: string): string;
}

export const userPreferences: UserPreferences = {
  baseSoundsPath: `/${APP_PREFIX_PATH}/sounds/`,
  tableRowHeightPx: 36,
  soundFileNotFound: (filename: string) => `Failed to load sound file "${filename}".`,
  configuredAudibleNotificationFileNotFound: (filename: string) =>
    `Failed to load sound file "${filename}" for configured audible notification. Sound will not play.`
};
