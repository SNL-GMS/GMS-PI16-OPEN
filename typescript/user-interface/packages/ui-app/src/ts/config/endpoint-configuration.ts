import { UI_URL } from '@gms/common-util';
import { AxiosRequestConfig } from 'axios';

export interface ServiceDefinition {
  readonly mockRequestConfig?: AxiosRequestConfig;
  readonly requestConfig: AxiosRequestConfig;
}

export interface RequestConfig {
  readonly [domain: string]: {
    readonly services: {
      readonly [serviceName: string]: ServiceDefinition;
    };
  };
}

// Default Config
export const defaultConfig: RequestConfig = {
  // Historical SOH Configuration
  historicalSoh: {
    services: {
      historicalByStationIdTimeAndSohMonitorTypes: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/ssam-control/retrieve-decimated-historical-station-soh`,
          headers: {
            // configure to receive msgpack encoded data
            accept: 'application/msgpack',
            'content-type': 'application/json'
          },
          // must specify that the response type is of type array buffer for
          // receiving and decoding msgpack data
          responseType: 'arraybuffer',
          timeout: 600000 // 10 minutes
        }
      }
    }
  }, // Station Definitions
  stationDefinition: {
    // Service endpoints for this component
    services: {
      getStationGroupsByNames: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/station-definition-service/station-definition/station-groups/query/names`,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          proxy: false,
          timeout: 60000
        }
      },
      getStations: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/station-definition-service/station-definition/stations/query/names`,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          proxy: false,
          timeout: 60000
        }
      },
      getStationsEffectiveAtTimes: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/station-definition-service/station-definition/stations/query/change-times`,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          proxy: false,
          timeout: 60000
        }
      }
    }
  },
  // System Messages Configuration
  systemMessage: {
    // Service endpoints for this component
    services: {
      getSystemMessageDefinitions: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/smds-service/retrieve-system-message-definitions`,
          headers: {
            accept: 'text/plain',
            'content-type': 'text/plain'
          },
          proxy: false,
          timeout: 60000,
          data: `"PlaceHolder"`
        }
      }
    }
  },
  // Processing Configuration Service
  processingConfiguration: {
    // Service endpoints for this component
    services: {
      getProcessingConfiguration: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/ui-processing-configuration-service/resolve`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          timeout: 60000
        }
      }
    }
  },
  // SSAM Control Service for SOH Configuration
  sohConfiguration: {
    // Service endpoints for this component
    services: {
      getSohConfiguration: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/ssam-control/retrieve-station-soh-monitoring-ui-client-parameters`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'text/plain'
          },
          timeout: 60000,
          data: `"PlaceHolder"`
        }
      }
    }
  },
  // User profile configuration
  userProfile: {
    // Service endpoints for this component
    services: {
      getUserProfile: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/user-manager-service/user-preferences`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'text/plain'
          },
          timeout: 60000,
          data: `"defaultUser"`
        }
      },
      setUserProfile: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/user-manager-service/user-preferences/store`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          timeout: 60000
        }
      }
    }
  },
  // Waveform Bridge Service Configuration
  waveformConfiguration: {
    // Service endpoints for this component
    services: {
      getWaveformConfiguration: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/waveform-manager-service/waveform/channel-segment/query/channel-timerange`,
          responseType: 'arraybuffer',
          proxy: false,
          headers: {
            accept: 'application/msgpack',
            'content-type': 'application/json'
          },
          timeout: 180000 // 3 mins until we figure out batching...
        }
      }
    }
  },
  // Workflow Service Configuration
  workflowConfiguration: {
    // Service endpoints for this component
    services: {
      workflow: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/workflow-manager-service/workflow-manager/workflow-definition`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'text/plain'
          },
          timeout: 180000, // 3 mins until we figure out batching...
          data: `"PlaceHolder"`
        }
      },
      stageIntervalsByIdAndTime: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/workflow-manager-service/workflow-manager/interval/stage/query/ids-timerange`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'text/plain'
          },
          timeout: 180000 // 3 mins until we figure out batching...
        }
      },
      updateActivityIntervalStatus: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/workflow-manager-service/workflow-manager/interval/activity/update`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          timeout: 60000
        }
      },
      updateInteractiveAnalysisStageIntervalStatus: {
        requestConfig: {
          method: 'post',
          url: `${UI_URL}/workflow-manager-service/workflow-manager/interval/stage/interactive-analysis/update`,
          responseType: 'json',
          proxy: false,
          headers: {
            accept: 'application/json',
            'content-type': 'application/json'
          },
          timeout: 60000
        }
      }
    }
  }
};
