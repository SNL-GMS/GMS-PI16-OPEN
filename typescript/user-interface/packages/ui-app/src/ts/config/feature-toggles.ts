interface ToggleOptions {
  readonly [id: string]: boolean | ToggleOptions;
}

export interface FeatureToggles extends ToggleOptions {
  readonly WAVEFORMS: {
    readonly LOAD_COLLAPSED_CHANNELS: boolean;
  };
}

/**
 * A collection of feature toggles that can be enabled/disabled to turn a feature on or off.
 */
export const FEATURE_TOGGLES: FeatureToggles = {
  WAVEFORMS: {
    /**
     * When true, will preload all collapsed child channelSegments in the waveform display.
     */
    LOAD_COLLAPSED_CHANNELS: false
  }
} as const;
