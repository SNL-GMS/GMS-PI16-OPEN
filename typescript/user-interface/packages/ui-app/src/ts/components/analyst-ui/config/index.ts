import { EnvironmentConfig, environmentConfig } from './environment-config';
import { SystemConfig, systemConfig } from './system-config';
import { UserPreferences, userPreferences } from './user-preferences';

export interface AnalystUiConfig {
  userPreferences: UserPreferences;
  environment: EnvironmentConfig;
  systemConfig: SystemConfig;
}

export const analystUiConfig: AnalystUiConfig = {
  userPreferences,
  environment: environmentConfig,
  systemConfig
};
export { EnvironmentConfig, environmentConfig } from './environment-config';
export { SystemConfig, systemConfig } from './system-config';
export { QcMaskDisplayFilters, UserPreferences, userPreferences } from './user-preferences';
