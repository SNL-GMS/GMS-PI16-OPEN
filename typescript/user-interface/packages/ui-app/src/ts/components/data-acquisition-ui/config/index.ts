import { DataAcquisitionUserPreferences, dataAcquisitionUserPreferences } from './user-preferences';

export interface DataAcquisitionUIConfig {
  dataAcquisitionUserPreferences: DataAcquisitionUserPreferences;
}

export const dataAcquisitionUIConfig: DataAcquisitionUIConfig = {
  dataAcquisitionUserPreferences
};

export { dataAcquisitionUserPreferences } from './user-preferences';
