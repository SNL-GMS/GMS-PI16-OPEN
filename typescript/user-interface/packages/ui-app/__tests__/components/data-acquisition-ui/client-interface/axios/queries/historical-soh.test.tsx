/* eslint-disable @typescript-eslint/no-magic-numbers */
import { SohTypes } from '@gms/common-model';
import Axios, { AxiosResponse } from 'axios';

import {
  handleHistoricalSohByStationQuery,
  historicalSohByStationConfig,
  historicalSohQuery,
  useHistoricalSohByStationQuery,
  withHistoricalSohByStationDefinitions
} from '../../../../../../src/ts/components/data-acquisition-ui/client-interface/axios/queries/historical-soh';
import { defaultConfig } from '../../../../../../src/ts/config/endpoint-configuration';

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

const sohMonitorType = SohTypes.SohMonitorType.LAG;
// eslint-disable-next-line @typescript-eslint/no-magic-numbers
const startTime = 1606818240000;
// eslint-disable-next-line @typescript-eslint/no-magic-numbers
const endTime = 1606818240100;

describe('Historical SOH Query', () => {
  it('is functions exported', () => {
    expect(historicalSohQuery).toBeDefined();
    expect(handleHistoricalSohByStationQuery).toBeDefined();
    expect(historicalSohByStationConfig).toBeDefined();
    expect(useHistoricalSohByStationQuery).toBeDefined();
    expect(withHistoricalSohByStationDefinitions).toBeDefined();
  });

  it('handle axios historical query throws', async () => {
    const data = {
      historicalSohInput: {
        startTime: 1,
        endTime: 5,
        samplesPerChannel: 5,
        sohMonitorType: SohTypes.SohMonitorType.LAG,
        stationName: 'station name'
      },
      maxQueryIntervalSize: 10000000
    };

    Axios.request = jest.fn().mockRejectedValue(new Error('connect ECONNREFUSED 127.0.0.1:80'));
    await expect(
      historicalSohQuery('key', {
        ...defaultConfig.historicalSoh.services.historicalByStationIdTimeAndSohMonitorTypes
          .requestConfig,
        data
      })
    ).rejects.toThrow('connect ECONNREFUSED 127.0.0.1:80');
  });

  it('handle axios query with no data response', async () => {
    const data = {
      historicalSohInput: {
        startTime: 1,
        endTime: 5,
        samplesPerChannel: 5,
        sohMonitorType: SohTypes.SohMonitorType.LAG,
        stationName: 'station name'
      },
      maxQueryIntervalSize: 10000000
    };

    const response: AxiosResponse<SohTypes.UiHistoricalSoh> = {
      status: 200,
      config: {},
      headers: {},
      statusText: '',
      data: {
        stationName: 'station name',
        calculationTimes: [],
        monitorValues: [],
        percentageSent: 0
      }
    };

    Axios.request = jest.fn().mockImplementation(async () => Promise.resolve(response));
    const result = await historicalSohQuery('key', {
      ...defaultConfig.historicalSoh.services.historicalByStationIdTimeAndSohMonitorTypes
        .requestConfig,
      data
    });
    expect(result).toMatchSnapshot();
  });

  it('handle axios query with data response', async () => {
    const data = {
      historicalSohInput: {
        startTime: 1,
        endTime: 5,
        samplesPerChannel: 5,
        sohMonitorType: SohTypes.SohMonitorType.LAG,
        stationName: 'station name'
      },
      maxQueryIntervalSize: 10000000
    };

    const response: AxiosResponse<SohTypes.UiHistoricalSoh> = {
      status: 200,
      config: {},
      headers: {},
      statusText: '',
      data: {
        calculationTimes: [0, 1, 2],
        stationName: `station name`,
        monitorValues: [
          {
            channelName: `channel name`,
            average: 6000.7,
            values: {
              values: [5000, 6000, 9000],
              type: SohTypes.SohValueType.DURATION
            }
          }
        ],
        percentageSent: 100
      }
    };

    Axios.request = jest.fn().mockImplementation(async () => Promise.resolve(response));
    const result = await historicalSohQuery('key', {
      ...defaultConfig.historicalSoh.services.historicalByStationIdTimeAndSohMonitorTypes
        .requestConfig,
      data
    });
    expect(result).toMatchSnapshot();
  });

  it('handle historical SOH query data - no data', async () => {
    const data: SohTypes.UiHistoricalSoh = {
      calculationTimes: [],
      stationName: `station name`,
      monitorValues: [],
      percentageSent: 100
    };

    const result = await handleHistoricalSohByStationQuery(0, 2, data);
    expect(result.stationName).toEqual('station name');
    expect(result.calculationTimes).toEqual([]);
    expect(result.minAndMax.xMin).toEqual(0);
    expect(result.minAndMax.xMax).toEqual(0);
    expect(result.minAndMax.yMin).toEqual(0);
    expect(result.minAndMax.yMax).toEqual(0);
    expect(result.monitorValues).toHaveLength(0);
  });

  it('handle historical SOH query data - duration', async () => {
    const data: SohTypes.UiHistoricalSoh = {
      calculationTimes: [0, 1, 2],
      stationName: `station name`,
      monitorValues: [
        {
          channelName: `channel name`,
          average: 6000.7,
          values: {
            values: [5000, 6000, 9000],
            type: SohTypes.SohValueType.DURATION
          }
        }
      ],
      percentageSent: 100
    };

    const result = await handleHistoricalSohByStationQuery(0, 2, data);
    expect(result.stationName).toEqual('station name');
    expect(result.calculationTimes).toEqual([0, 0.001, 0.002]);
    expect(result.minAndMax.xMin).toEqual(0);
    expect(result.minAndMax.xMax).toEqual(0.002);
    expect(result.minAndMax.yMin).toEqual(5);
    expect(result.minAndMax.yMax).toEqual(9);
    expect(result.monitorValues).toHaveLength(1);
    expect(result.monitorValues[0].channelName).toEqual('channel name');
    expect(result.monitorValues[0].type).toEqual(SohTypes.SohValueType.DURATION);
    expect(result.monitorValues[0].average).toEqual(6);
    expect(result.monitorValues[0].values).toEqual(new Float32Array([0, 5, 50, 6, 100, 9]));
  });

  it('handle historical SOH query data - percent', async () => {
    const data: SohTypes.UiHistoricalSoh = {
      calculationTimes: [0, 1, 2],
      stationName: `station name`,
      monitorValues: [
        {
          channelName: `channel name`,
          average: 7,
          values: {
            values: [5, 6, 9],
            type: SohTypes.SohValueType.PERCENT
          }
        }
      ],
      percentageSent: 100
    };

    const result = await handleHistoricalSohByStationQuery(0, 2, data);
    expect(result.stationName).toEqual('station name');
    expect(result.calculationTimes).toEqual([0, 0.001, 0.002]);
    expect(result.minAndMax.xMin).toEqual(0);
    expect(result.minAndMax.xMax).toEqual(0.002);
    expect(result.minAndMax.yMin).toEqual(5);
    expect(result.minAndMax.yMax).toEqual(9);
    expect(result.monitorValues).toHaveLength(1);
    expect(result.monitorValues[0].channelName).toEqual('channel name');
    expect(result.monitorValues[0].type).toEqual(SohTypes.SohValueType.PERCENT);
    expect(result.monitorValues[0].average).toEqual(7);
    expect(result.monitorValues[0].values).toEqual(new Float32Array([0, 5, 50, 6, 100, 9]));
  });

  it('is config set correctly', () => {
    const input: SohTypes.UiHistoricalSohInput = {
      startTime,
      endTime,
      sohMonitorType,
      samplesPerChannel: 50000,
      stationName: 'AAK'
    };
    const config: any = historicalSohByStationConfig(input, 1000000);

    expect(config.queryKey).toHaveLength(2);
    expect(config.queryKey[0]).toEqual('historicalByStationIdTimeAndSohMonitorTypes');
    expect(config.queryKey[1].method).toEqual('post');
    expect(config.queryKey[1].headers).toEqual({
      accept: 'application/msgpack',
      'content-type': 'application/json'
    });
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    expect(config.queryKey[1].timeout).toEqual(600000);
  });

  it('is creating the config correctly for lag', () => {
    const config = historicalSohByStationConfig(
      {
        samplesPerChannel: 50000,
        startTime,
        endTime,
        sohMonitorType: SohTypes.SohMonitorType.LAG,
        stationName: `station name`
      },
      1000000
    );

    expect(config).toMatchSnapshot();
  });

  it('is creating the config correctly for timeliness', () => {
    const config = historicalSohByStationConfig(
      {
        samplesPerChannel: 50000,
        startTime,
        endTime,
        sohMonitorType: SohTypes.SohMonitorType.TIMELINESS,
        stationName: `station name`
      },
      1000000
    );

    expect(config).toMatchSnapshot();
  });

  it('is creating the config correctly for missing', () => {
    const config = historicalSohByStationConfig(
      {
        samplesPerChannel: 50000,
        startTime,
        endTime,
        sohMonitorType: SohTypes.SohMonitorType.MISSING,
        stationName: `station name`
      },
      1000000
    );

    expect(config).toMatchSnapshot();
  });
});
