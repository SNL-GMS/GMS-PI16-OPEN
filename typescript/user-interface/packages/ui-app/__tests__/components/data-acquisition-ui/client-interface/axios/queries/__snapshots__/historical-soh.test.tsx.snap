// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Historical SOH Query handle axios query with data response 1`] = `
Object {
  "calculationTimes": Array [
    0,
    0.001,
    0.002,
  ],
  "minAndMax": Object {
    "xMax": 0.002,
    "xMin": 0,
    "yMax": 9,
    "yMin": 5,
  },
  "monitorValues": Array [
    Object {
      "average": 6,
      "channelName": "channel name",
      "type": "DURATION",
      "values": Float32Array [
        -25,
        5,
        0,
        6,
        25,
        9,
      ],
    },
  ],
  "percentageSent": 100,
  "stationName": "station name",
}
`;

exports[`Historical SOH Query handle axios query with no data response 1`] = `
Object {
  "calculationTimes": Array [],
  "minAndMax": Object {
    "xMax": Infinity,
    "xMin": -Infinity,
    "yMax": Infinity,
    "yMin": -Infinity,
  },
  "monitorValues": Array [],
  "percentageSent": 0,
  "stationName": "station name",
}
`;

exports[`Historical SOH Query is creating the config correctly for lag 1`] = `
Object {
  "config": Object {
    "refetchOnWindowFocus": false,
    "staleTime": Infinity,
  },
  "queryFn": [Function],
  "queryKey": Array [
    "historicalByStationIdTimeAndSohMonitorTypes",
    Object {
      "data": Object {
        "historicalSohInput": Object {
          "endTime": 1606818240100,
          "samplesPerChannel": 50000,
          "sohMonitorType": "LAG",
          "startTime": 1606818240000,
          "stationName": "station name",
        },
        "maxQueryIntervalSize": 1000000,
      },
      "headers": Object {
        "accept": "application/msgpack",
        "content-type": "application/json",
      },
      "method": "post",
      "responseType": "arraybuffer",
      "timeout": 600000,
      "url": "http://localhost/ssam-control/retrieve-decimated-historical-station-soh",
    },
  ],
}
`;

exports[`Historical SOH Query is creating the config correctly for missing 1`] = `
Object {
  "config": Object {
    "refetchOnWindowFocus": false,
    "staleTime": Infinity,
  },
  "queryFn": [Function],
  "queryKey": Array [
    "historicalByStationIdTimeAndSohMonitorTypes",
    Object {
      "data": Object {
        "historicalSohInput": Object {
          "endTime": 1606818240100,
          "samplesPerChannel": 50000,
          "sohMonitorType": "MISSING",
          "startTime": 1606818240000,
          "stationName": "station name",
        },
        "maxQueryIntervalSize": 1000000,
      },
      "headers": Object {
        "accept": "application/msgpack",
        "content-type": "application/json",
      },
      "method": "post",
      "responseType": "arraybuffer",
      "timeout": 600000,
      "url": "http://localhost/ssam-control/retrieve-decimated-historical-station-soh",
    },
  ],
}
`;

exports[`Historical SOH Query is creating the config correctly for timeliness 1`] = `
Object {
  "config": Object {
    "refetchOnWindowFocus": false,
    "staleTime": Infinity,
  },
  "queryFn": [Function],
  "queryKey": Array [
    "historicalByStationIdTimeAndSohMonitorTypes",
    Object {
      "data": Object {
        "historicalSohInput": Object {
          "endTime": 1606818240100,
          "samplesPerChannel": 50000,
          "sohMonitorType": "TIMELINESS",
          "startTime": 1606818240000,
          "stationName": "station name",
        },
        "maxQueryIntervalSize": 1000000,
      },
      "headers": Object {
        "accept": "application/msgpack",
        "content-type": "application/json",
      },
      "method": "post",
      "responseType": "arraybuffer",
      "timeout": 600000,
      "url": "http://localhost/ssam-control/retrieve-decimated-historical-station-soh",
    },
  ],
}
`;
